src/amo/components/Link.js

```diff
@@ -9,37 +9,75 @@ import { Link } from 'react-router';
 
 export class LinkBase extends React.Component {
   static propTypes = {
-    base: PropTypes.string,
+    clientApp: PropTypes.string.isRequired,
     children: PropTypes.node,
     href: PropTypes.string,
+    lang: PropTypes.string.isRequired,
+    prependClientApp: PropTypes.bool,
+    prependLang: PropTypes.bool,
     to: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
   }
 
   static defaultProps = {
-    base: '',
+    prependClientApp: true,
+    prependLang: true,
+  }
+
+  urlPrefix({ clientApp, lang, prependClientApp, prependLang } = {}) {
+    if (prependClientApp && prependLang) {
+      return `/${lang}/${clientApp}/`;
+    } else if (prependClientApp) {
+      return `/${clientApp}/`;
+    } else if (prependLang) {
+      return `/${lang}/`;
+    }
+
+    // If no prefixes should be applied we'll return null and pass the
+    // path through to an <a> tag or <Link> component.
+    return null;
   }
 
   render() {
-    const { base, children, href, to, ...customProps } = this.props;
+    const {
+      clientApp,
+      children,
+      href,
+      lang,
+      prependClientApp,
+      prependLang,
+      to,
+      ...customProps
+    } = this.props;
+    const urlPrefix = this.urlPrefix({
+      clientApp, lang, prependClientApp, prependLang });
 
     if (typeof href === 'string' && typeof to !== 'undefined') {
       throw new Error(
         'Cannot use "href" prop and "to" prop in the same Link component');
     }
 
+    if (
+      typeof to !== 'undefined' && (
+        (typeof to === 'string' && !to.startsWith('/')) ||
+        (to && to.pathname && !to.pathname.startsWith('/'))
+      )
+    ) {
+      throw new Error(
+        '"to" prop cannot contain a relative path; it must start with a "/".');
+    }
+
     if (typeof href === 'string') {
-      const linkHref = href.startsWith('/') ? path.join(base, href) : href;
+      const linkHref = urlPrefix ? path.join(urlPrefix, href) : href;
```

Comment:
I don't think this should have ever used `path.join()` because that will use back slashes on Windows. It should always join a URL with forward slashes -- maybe with [`url.resolve()`](https://nodejs.org/api/url.html#url_url_resolve_from_to)?